### PR TITLE
Use rustdoc.css for error index

### LINKED ISSUE #86512: Use rustdoc files again for error index page
#86503 made a small fix for the error index page, but it's still very likely that it'll break in no time because of it's not using rustdoc's style directly. Instead, we should keep specific style for this page in `rust.css` and use the "normal" rustdoc style for the rest.

### PR DESCRIPTION
**Pull request by fee1-dead**
Closes #86512.

### CODE REVIEW

**Comment by ehuss** [comment]
Note that `rust.css` is used for a bunch of other pages, and this breaks a lot of the formatting for them.  The primary one is the main offline landing page, but it is used for a few others. Please check all the things in [doc.rs](https://github.com/rust-lang/rust/blob/master/src/bootstrap/doc.rs) that use it.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
This looks good to me, thanks for this fix! I'd feel better though if the @rust-lang/infra team took a look to this PR to ensure something wasn't overlooked.

**Comment by jyn514** [comment]
@GuillaumeGomez I don't understand why you pinged the infra team. They don't own the index page AFAIK, I'm not sure who does.

@ehuss you meant doc.rust-lang.org, right? @fee1-dead @GuillaumeGomez  can one of you follow up on that? You can build all the docs locally with `x.py doc --stage 1`.

**Comment by GuillaumeGomez** [comment]
Last time I proposed to change the style on the error index page, it was rejected. (I think we discussed about it with @pietroalbini but don't remember...)

I already checked it locally and it works so if there is no other thing to check, it can be r+ed.

**Comment by ehuss** [comment]
> @ehuss you meant doc.rust-lang.org, right?

The link is to https://github.com/rust-lang/rust/blob/master/src/bootstrap/doc.rs, where the usage of `rust.css` is.

**Comment by pietroalbini** [comment]
> Last time I proposed to change the style on the error index page, it was rejected. (I think we discussed about it with @pietroalbini but don't remember...)

@GuillaumeGomez do you recall when or where? I don't remember anything about the index page.

**Comment by GuillaumeGomez** [comment]
I don't remember at all and can't find the PR. Maybe I just dreamed all this...

File: src/bootstrap/doc.rs [hunk 1]
```diff
@@ -374,9 +374,15 @@ impl Step for Standalone {
             }
 
             if filename == "not_found.md" {
-                cmd.arg("--markdown-css").arg("https://doc.rust-lang.org/rust.css");
+                cmd.arg("--markdown-css")
+                    .arg(format!("https://doc.rust-lang.org/rustdoc{}.css", &builder.version))
+                    .arg("--markdown-css")
+                    .arg("https://doc.rust-lang.org/rust.css");
```
Comment (jyn514) [inline src/bootstrap/doc.rs:380]: This doesn't seem right, that file will go away as soon as this PR is merged. Why was it necessary? That makes me think the whole approach isn't feasible.

Comment (jyn514) [reply src/bootstrap/doc.rs:380]: Oh wait I misunderstood what's going on - you're removing *most* of rust.css and only leaving the things specific to the website. Ignore me.

**Comment by jyn514** [comment]
@bors r=GuillaumeGomez

This is easy enough to revert if necessary and it seems worthwhile to not have to keep things in sync.

**Pull request merged**